projects/kit/components/tabs/tabs.directive.ts

```diff
@@ -65,4 +87,33 @@ export class TuiTabsDirective implements AfterViewChecked {
             nativeElement.setAttribute(`tabIndex`, active ? `0` : `-1`);
         });
     }
+
+    ngOnDestroy(): void {
+        this.windowRef.clearInterval(this.scrollingId);
+    }
+
+    private safelyScrollToActiveElement(): void {
+        this.ngZone.runOutsideAngular(() => {
+            this.windowRef.clearTimeout(this.scrollingId);
+            this.scrollingId = this.windowRef.setTimeout(
+                () => this.scrollToVisibleAreaActiveElement(),
+                this.autoScrollTimeout,
+            );
+        });
+    }
+
+    private scrollToVisibleAreaActiveElement(): void {
+        const tab = this.activeElement;
+        const tabs = tab?.parentElement;
+
+        if (!tuiIsHTMLElement(tab) || !tuiIsHTMLElement(tabs)) {
+            return;
+        }
+
+        const invisibleInLeftArea = tab.offsetLeft > tabs.offsetWidth + tabs.scrollLeft;
+
+        if (invisibleInLeftArea) {
```

Comment:
What if it's invisible on the other side?

Reply:
fixed